clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/s3/S3ArtifactAccount.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.artifacts.s3;
+
+import com.netflix.spinnaker.clouddriver.artifacts.config.ArtifactAccount;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class S3ArtifactAccount extends ArtifactAccount
+{
+  private String name = "s3-account";
```

Comment:
Ok, following up here. I want to make sure I understand before implementing it. So the behavior of the code as is will be to use the default credential provider, resolving whatever it can from the instance, the env, or configuration, etc. You're saying we could add a list of roles to assume, adding those as potential credentials to use?

Reply:
I'm not familiar enough with AWS creds to give a firm answer, but I know that AWS accounts (even on running on GCE) can assume control over multiple other accounts using `accountId` and `assumeRule` like done for the AWS provider. https://github.com/spinnaker/halyard/blob/master/halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/aws/AwsAccount.java#L36

Maybe we can implement this as is, and then defer to @imosquera to implement the multi-account logic?